packages/snippet-bot/src/snippet-bot.ts

```diff
@@ -143,376 +143,392 @@ async function getFiles(dir: string, allFiles: string[]) {
   return allFiles;
 }
 
-export = (app: Application) => {
-  app.on(
-    [
-      'issues.opened',
-      'issues.reopened',
-      'pull_request.opened',
-      'pull_request.reopened',
-      'pull_request.edited',
-      'pull_request.labeled',
-      'pull_request.synchronize',
-    ],
-    async context => {
-      const repoUrl = context.payload.repository.full_name;
-      const defaultBranch = context.payload.repository.default_branch;
-      let configOptions!: ConfigurationOptions | null;
-      try {
-        configOptions = await context.config<ConfigurationOptions>(
-          CONFIGURATION_FILE_PATH
-        );
-      } catch (err) {
-        err.message = `Error reading configuration: ${err.message}`;
-        logger.error(err);
-        // Now this bot is only enabled if it finds the configuration file.
-        // Exiting.
-        return;
-      }
+async function getConfigOptions(
+  context: Context
+): Promise<ConfigurationOptions | null> {
+  let configOptions: ConfigurationOptions | null = null;
+  try {
+    configOptions = await context.config<ConfigurationOptions>(
+      CONFIGURATION_FILE_PATH
+    );
+  } catch (err) {
+    err.message = `Error reading configuration: ${err.message}`;
+    logger.error(err);
+  }
+  return configOptions;
+}
 
-      if (configOptions === null) {
-        logger.info(`snippet-bot is not configured for ${repoUrl}.`);
-        return;
-      }
-      const configuration = new Configuration({
-        ...DEFAULT_CONFIGURATION,
-        ...configOptions,
+async function fullScan(context: Context, configuration: Configuration) {
+  const installationId = context.payload.installation.id;
+  const commentMark = `<!-- probot comment [${installationId}]-->`;
+  const owner = context.payload.repository.owner.login;
+  const repo = context.payload.repository.name;
+  const defaultBranch = context.payload.repository.default_branch;
+
+  if (context.payload.issue?.title.includes(FULL_SCAN_ISSUE_TITLE)) {
+    // full scan start
+    const issueNumber = context.payload.issue.number;
```

Comment:
Consider inverting this check and returning early, so you can get rid of one level of indentation.

```suggestion
  if (!context.payload.issue?.title.includes(FULL_SCAN_ISSUE_TITLE)) {
    return;
  }
  const issueNumber = context.payload.issue.number;
  // ...
```

Reply:
Good suggestion. Will do.

Reply:
Done